### PR TITLE
[improve](s3) s3 load provider should be case insensitive

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Some S3 vendors strictly require upper case provider info.
Provider properties in broker load `WITH S3` statement should be case insensitive.

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CopyStmt.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ private void analyzeStagePB(StagePB stagePB) throws AnalysisException {
         }
         brokerProperties.put(S3_BUCKET, objInfo.getBucket());
         brokerProperties.put(S3_PREFIX, objInfo.getPrefix());
-        brokerProperties.put(S3Properties.PROVIDER, objInfo.getProvider().toString());
+        brokerProperties.put(S3Properties.PROVIDER, objInfo.getProvider().toString().toUpperCase());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CopyStmt.java:208]: please add NOTICE comment to these upper case change.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java [hunk 2]
```diff
@@ -527,7 +527,7 @@ private String getProviderFromEndpoint() {
         Map<String, String> properties = brokerDesc.getProperties();
         for (Map.Entry<String, String> entry : properties.entrySet()) {
             if (entry.getKey().equalsIgnoreCase(S3Properties.PROVIDER)) {
-                return entry.getValue();
+                return entry.getValue().toUpperCase();
```
Comment (kaijchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java:530]:
```suggestion
                // S3 Provider properties should be case insensitive.
                return entry.getValue().toUpperCase();
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/StageProperties.java [hunk 3]
```diff
@@ -135,7 +135,7 @@ private void analyzeStorageProperties() throws AnalysisException {
         }
         properties.put(PREFIX, prefix);
         // analyze provider
-        String provider = properties.get(PROVIDER);
+        String provider = properties.get(PROVIDER).toUpperCase();
```
Comment (kaijchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/StageProperties.java:138]:
```suggestion
        // S3 Provider properties should be case insensitive.
        String provider = properties.get(PROVIDER).toUpperCase();
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java [hunk 4]
```diff
@@ -197,7 +197,7 @@ public static Map<String, String> requiredS3TVFProperties(Map<String, String> pr
     private static void checkProvider(Map<String, String> properties) throws DdlException {
         if (properties.containsKey(PROVIDER)) {
             properties.put(PROVIDER, properties.get(PROVIDER).toUpperCase());
-            if (!PROVIDERS.stream().anyMatch(s -> s.equals(properties.get(PROVIDER)))) {
+            if (!PROVIDERS.stream().anyMatch(s -> s.equals(properties.get(PROVIDER).toUpperCase()))) {
```
Comment (kaijchen) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:200]:
```suggestion
            // S3 Provider properties should be case insensitive.
            if (!PROVIDERS.stream().anyMatch(s -> s.equals(properties.get(PROVIDER).toUpperCase()))) {
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java [hunk 5]
```diff
@@ -330,7 +330,7 @@ public static Cloud.ObjectStoreInfoPB.Builder getObjStoreInfoPB(Map<String, Stri
             builder.setExternalEndpoint(properties.get(S3Properties.EXTERNAL_ENDPOINT));
         }
         if (properties.containsKey(S3Properties.PROVIDER)) {
-            builder.setProvider(Provider.valueOf(properties.get(S3Properties.PROVIDER)));
+            builder.setProvider(Provider.valueOf(properties.get(S3Properties.PROVIDER).toUpperCase()));
```
Comment (kaijchen) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/constants/S3Properties.java:333]:
```suggestion
            // S3 Provider properties should be case insensitive.
            builder.setProvider(Provider.valueOf(properties.get(S3Properties.PROVIDER).toUpperCase()));
```

File: fe/fe-core/src/test/java/org/apache/doris/analysis/StageTest.java [hunk 6]
```diff
@@ -103,7 +103,7 @@ public void testAnalyzeStorageProperties() throws Exception {
                 + "'prefix' = 'tmp_prefix', "
                 + "'provider' = 'abc', "
                 + "'ak'='tmp_ak', 'sk'='tmp_sk', 'access_type'='aksk');";
-        parseAndAnalyzeWithException(sql, "Property provider with invalid value abc");
+        parseAndAnalyzeWithException(sql, "Property provider with invalid value ABC");
```
Comment (kaijchen) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/StageTest.java:106]:
```suggestion
        // S3 Provider will be converted to upper case.
        parseAndAnalyzeWithException(sql, "Property provider with invalid value ABC");
```

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/Config.groovy [hunk 7]
```diff
@@ -630,7 +630,7 @@ class Config {
             } else if (config.s3Source == "huawei") {
                 s3Provider = "OBS"
             } else if (config.s3Source == "azure") {
-                s3Provider = "AZURE"
+                s3Provider = "Azure"
```
Comment (kaijchen) [inline regression-test/framework/src/main/groovy/org/apache/doris/regression/Config.groovy:633]:
```suggestion
                s3Provider = "Azure" // case insensitive test
```

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**